packages/adapters/bitcoin/src/adapter.ts

```diff
@@ -41,17 +48,58 @@ export class BitcoinAdapter extends AdapterBlueprint<BitcoinConnector> {
       throw new Error('connectionControllerClient:connectExternal - connector is undefined')
     }
 
-    this.connector = connector
-    this.bindEvents(this.connector)
-
-    const address = await connector.connect()
-
     const chain = connector.chains.find(c => c.id === params.chainId) || connector.chains[0]
 
     if (!chain) {
       throw new Error('The connector does not support any of the requested chains')
     }
 
+    const connection = this.connections.find(c => c.connectorId === connector.id)
```

Comment:
Pretty much implementation went into the adapter clients - this feels like went away from our approach abstracting things with the AdapterBlueprint and keep the clients simple as possible. Like to ask if it's possible to simplify these logics in Blueprint and not repeat on each adapter? From the point of the review it's a bit hard to see so like to discuss.

Also ping for @tomiir @svenvoskamp